vector/src/main/java/im/vector/app/core/extensions/TextInputLayout.kt

```diff
@@ -30,3 +34,26 @@ fun TextInputLayout.hasSurroundingSpaces() = editText().text.toString().let { it
 fun TextInputLayout.hasContentFlow(mapper: (CharSequence) -> CharSequence = { it }) = editText().textChanges().map { mapper(it).isNotEmpty() }
 
 fun TextInputLayout.content() = editText().text.toString()
+
+fun TextInputLayout.hasContent() = !editText?.text.isNullOrEmpty()
+
+fun TextInputLayout.associateContentStateWith(button: View) {
+    editText?.addTextChangedListener(object : SimpleTextWatcher() {
+        override fun afterTextChanged(s: Editable) {
+            val newContent = s.toString()
+            button.isEnabled = newContent.isNotEmpty()
+        }
+    })
+}
+
+fun TextInputLayout.setOnImeDone(action: () -> Unit) {
```

Comment:
happy to rename for consistency, with [android-ktx](https://developer.android.com/kotlin/ktx/extensions-list) the convention is turning into `doOnX`, although this favours `add` rather than setters

Reply:
a712e1f186bd5ae8bb53e203d983d1dbc10ac587